### PR TITLE
Initial NPU C++ Example

### PR DESCRIPTION
**Pull request by rnwang04**
### 1. Why the change?

https://github.com/analytics-zoo/nano/issues/1716
First version example of our current C++ NPU solution. 

### 2. User API changes

See example README.md .

### 3. Summary of the change 

- Provide convert script
- Provide cmake build script and related cpp source code
- Initial README

### 4. How to test?
- [ ] Unit test: Please manually trigger the PR Validation [here](https://github.com/intel-analytics/ipex-llm-workflow/actions/workflows/llm-PR-validation.yml) by inputting the PR number (e.g., `1234`). And paste your action link here once it has been successfully finished.

### CODE REVIEW

File: python/llm/example/NPU/HF-Transformers-AutoModels/LLM/CPP_Examples/README.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# C++ Example of running HuggingFace `transformers` Models on Intel NPU using IPEX-LLM
+In this directory, you will find a C++ example on how to run HuggingFace `transformers` LLM models on Intel NPUs (leveraging *Intel NPU Acceleration Library*). See the table blow for verified models.
```
Comment (jason-dai) [inline python/llm/example/NPU/HF-Transformers-AutoModels/LLM/CPP_Examples/README.md:2]: a C++ example on how to run LLM models on Intel NPUs using IPEX-LLM

File: python/llm/example/NPU/HF-Transformers-AutoModels/LLM/CPP_Examples/README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
+# C++ Example of running HuggingFace `transformers` Models on Intel NPU using IPEX-LLM
```
[the remark this answers is not in the document]
Comment (rnwang04) [reply python/llm/example/NPU/HF-Transformers-AutoModels/LLM/CPP_Examples/README.md]: update to this.

File: python/llm/example/NPU/HF-Transformers-AutoModels/LLM/CPP_Examples/README.md (hunk 1 above)
Comment (rnwang04) [reply python/llm/example/NPU/HF-Transformers-AutoModels/LLM/CPP_Examples/README.md]: update to this.

File: python/llm/example/NPU/HF-Transformers-AutoModels/LLM/CPP_Examples/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        ${LIBRARY_DIR}/npu_llm.dll
+        ${CMAKE_BINARY_DIR}/Release/
+    COMMENT "Copying npu_llm.dll to build/Release\n"
+)
+
+add_custom_command(TARGET llm-npu-cli POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E copy_directory
+        ${DLL_DIR}/
+        ${CMAKE_BINARY_DIR}/Release/
+    COMMENT "Copying ov dll to build/Release\n"
```
Comment (jason-dai) [inline python/llm/example/NPU/HF-Transformers-AutoModels/LLM/CPP_Examples/CMakeLists.txt:38]: Copying dependency to build/Release\n

Comment (rnwang04) [reply python/llm/example/NPU/HF-Transformers-AutoModels/LLM/CPP_Examples/CMakeLists.txt]: done

**Review by jason-dai — APPROVED** [review]
LGTM

**Pull request merged**